lib/hyrax/specs/shared_specs/indexers.rb

```diff
@@ -3,7 +3,8 @@
 RSpec.shared_examples 'a Hyrax::Resource indexer' do
   subject(:indexer) { indexer_class.new(resource: resource) }
   let(:ids)         { ['id1', 'id2'] }
-  let(:resource)    { Hyrax::Resource.new(alternate_ids: ids) }
+  let(:the_resource_class) { defined?(resource_class) ? resource_class : Hyrax::Work }
```

Comment:
Setting resource_class to Hyrax::Work when it isn't defined allows for backward compatibility.

Reply:
Passing in the resource_class allows #to_solr to successfully create the solr_doc when the indexer is expecting the resource to have custom metadata defined.